packaging/docker/build.sh

```diff
@@ -1,82 +1,56 @@
 #!/bin/bash
-# SPDX-License-Identifier: GPL-3.0-or-later
-# Author  : Pawel Krupa (paulfantom)
 # Cross-arch docker build helper script
-# Needs docker in version >18.02 due to usage of manifests
+#
+# Copyright: SPDX-License-Identifier: GPL-3.0-or-later
+#
+# Author  : Pawel Krupa (paulfantom)
+# Author  : Pavlos Emm. Katsoulakis (paul@netdata.cloud)
 
 set -e
 
-if [ ! -f .gitignore ]; then
-	echo "Run as ./packaging/docker/$(basename "$0") from top level directory of git repository"
-	exit 1
-fi
+VERSION="$1"
+REPOSITORY="${REPOSITORY:-netdata}"
+declare -A ARCH_MAP
+ARCH_MAP=( ["i386"]="386" ["amd64"]="amd64" ["armhf"]="arm" ["aarch64"]="arm64")
+DEVEL_ARCHS=(amd64)
+ARCHS="${!ARCH_MAP[@]}"
 
-if [ "$1" == "" ]; then
-    VERSION=$(git tag --points-at)
-else
-    VERSION="$1"
+# When development mode is set, build on DEVEL_ARCHS
+if [ ! -z ${DEVEL+x} ]; then
+    declare -a ARCHS=(${DEVEL_ARCHS[@]})
 fi
+
+# Ensure there is a version, the most appropriate one
 if [ "${VERSION}" == "" ]; then
-    VERSION="latest"
+    VERSION=$(git tag --points-at)
+    if [ "${VERSION}" == "" ]; then
+        VERSION="latest"
+    fi
 fi
 
-declare -A ARCH_MAP
-ARCH_MAP=( ["i386"]="386" ["amd64"]="amd64" ["armhf"]="arm" ["aarch64"]="arm64")
-if [ -z ${DEVEL+x} ]; then
-    declare -a ARCHITECTURES=(i386 armhf aarch64 amd64)
-else
-    declare -a ARCHITECTURES=(amd64)
-    unset DOCKER_PASSWORD
-    unset DOCKER_USERNAME
+# TODO: Need a more stable way to find where to run from.
```

Comment:
You can use this if you want: https://github.com/openshift/machine-api-operator/blob/master/hack/go-build.sh#L15-L16